src/containers/SignUp/SignUp.js

```diff
@@ -52,41 +96,87 @@ class SignUp extends Component {
-                {this.validator.message('email', this.state.email, 'required|email')}
+                <input
+                  className="field"
+                  name="email"
+                  type="email"
+                  onChange={this.handleUserInput}
+                />
+                {this.validator.message(
+                  "email",
+                  this.state.email,
+                  "required|email"
+                )}
               </div>
               <div>
-                <label className="labels vertical-spacing">Password <span className="required-indicator">*</span></label>
+                <label className="labels vertical-spacing">
+                  Password <span className="required-indicator">*</span>
+                </label>
                 <br />
-                <input className="field" name="password" type="password" onChange={this.handleUserInput}/>
-                {this.validator.message('password', this.state.password, 'required|min:6|max:20')}
+                <input
+                  className="field"
+                  name="password"
+                  type="password"
+                  onChange={this.handleUserInput}
+                />
+                {this.validator.message(
+                  "password",
+                  this.state.password,
+                  "required|min:6|max:20"
+                )}
               </div>
               <div>
-                <label className="labels vertical-spacing">Confirm Password <span className="required-indicator">*</span></label>
+                <label className="labels vertical-spacing">
+                  Confirm Password <span className="required-indicator">*</span>
+                </label>
                 <br />
-                <input className="field" name="confirmPassword" type="password" onChange={this.handleUserInput}/>
-                {this.validator.message('confirmPassword', this.state.confirmPassword, 'requried|cp:'+this.state.password)}
+                <input
+                  className="field"
+                  name="confirmPassword"
+                  type="password"
+                  onChange={this.handleUserInput}
+                />
+                {this.validator.message(
+                  "confirmPassword",
+                  this.state.confirmPassword,
+                  "requried|cp:" + this.state.password
+                )}
               </div>
               <div className="term-conditions">
                 By clicking 'Create account', you agree to our Terms and
                 Conditions and acknowledge that you have read our Privacy Policy
                 and Disclosure Guidelines.
               </div>
               <div>
-                <button className="button" name="signUpBtn" type="submit">
+                <button
+                  className="button"
+                  name="signUpBtn"
+                  type="submit"
+                  onClick={this.postData}
```

Comment:
@Humnaa there is already a submit event bind with the form i.e. `submitForm`. So instead of binding a new click event you can call api inside that very same method.

cc: @abdulhaqshah

Reply:
Extra function has removed and write functionality in existing function "submitForm"